samples/subsys/fs/littlefs/src/main.c

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright (c) 2019 Peter Bigot Consulting, LLC
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/* Sample which uses the filesystem API with littlefs */
+
+#include <stdio.h>
+
+#include <zephyr.h>
+#include <device.h>
+#include <fs/fs.h>
+#include <fs/littlefs.h>
+#include <storage/flash_map.h>
+
+/* Matches LFS_NAME_MAX */
+#define MAX_PATH_LEN 255
+
+static struct fs_littlefs lfs_storage;
+static struct fs_mount_t lfs_storage_mnt = {
+	.type = FS_LITTLEFS,
+	.fs_data = &lfs_storage,
+	.storage_dev = (void *)DT_FLASH_AREA_STORAGE_ID,
+	.mnt_point = "/lfs",
+};
+
+void main(void)
+{
+	struct fs_mount_t *mp = &lfs_storage_mnt;
+	unsigned int id = (uintptr_t)mp->storage_dev;
```

Comment:
Should this not be a cast `unsigned int`?

Reply:
No.  It's only valid to cast `void *` to or from `[u]intptr_t`: direct cast to `unsigned int` would be incorrect if pointers require a larger underlying type.

There could be an explicit cast to truncate to the destination type, though it's not required by C for unsigned integral types.